### DISCUSSION TITLE
Not able to upload files with flutter and react native

### DISCUSSION THREAD

**Discussion by aryabyte21** [pocketbase/pocketbase#1329] [Q&A]
Hey community, I am not able to upload the pdf files to pocketbase I tried a lot if someone has done this before can u support me with some example code or perhaps redirect me to some resource. I am really looking forward to creating an app for sharing notes and monetizing it the frontend is ready only issue I am facing is uploading the pdf file please help me. HELP NEEDED.

**Comment by ganigeorgiev** [comment]
I don't see what Flutter has to do with React native and I'm not sure how you are combining them both in your app.

You can find both JS and Dart file upload examples in https://pocketbase.io/docs/files-handling/#uploading-files.


For Dart, you have to pass a [`http.MultipartFile` instance](https://pub.dev/documentation/http/latest/http/MultipartFile-class.html). This instance can be created from a raw file string content, bytes List or file path.

For JS, you need to provide the request body as [`FormData` object](https://developer.mozilla.org/en-US/docs/Web/API/FormData).


If you need further help, please a provide a code sample how you are sending the request(s).

**Reply by aryabyte21** [reply]
i'm sorry flutter code came thrice there was some network issue

**Comment by aryabyte21** [comment]
## React Native
```
const [pdfFile, setPdfFile] = useState<string | null>(null)
  const [pdfFile1, setPdfFile1] = useState<File>(null)
  const [size, setSize] = useState<number | null>(null)

  async function handleSelectPdf() {
    try {
      const res = await DocumentPicker.getDocumentAsync({
        type: "application/pdf",
      })
      if (res.type === "success") {
        setPdfFile(res.name)
        setPdfFile1(res.file)
        setSize(res.size)
        console.log(res.output)
      }
    } catch (error) {
      console.error(error)
    }
  }

const uploadPdf = async () => {
 const fileUri = pdfFile
    const getFile = async () => {
      try {
        const fileInfo = await FileSystem.getInfoAsync(fileUri)
        console.log(fileInfo)
      } catch (error) {
        console.error(error)
      }
    }

    const pb = new PocketBase("http://127.0.0.1:8090")
    const formData = new FormData()
    formData.append('subject', "hello13")
    formData.append('document', pdfFile1)

    const record = await pb.collection("notes").create(formData)
    console.log(formData)
  }
}
 ```

**Comment by aryabyte21** [comment]
## Flutter
```
 TextButton(
                        child: Text('Select file'),
                        onPressed: () async {
                          String? filePath = '/Users/aryaarun/Desktop/Blog.pdf';
                          Uint8List fileData = Uint8List(0);
                          FilePickerResult? result1;

                          // ...
                          // Inside the onPressed callback of the button
                          try {
                            result1 = await FilePicker.platform.pickFiles();
                            filePath = result1!.files.first.path;
                          } catch (e) {
                            print(e.toString());
                          }
                          print(filePath.toString());
                          try {
                            final file = File(filePath!);
                            fileData = file.readAsBytesSync();
                          } catch (e) {
                            print(e.toString());
                          }

                          // Use the filePath variable to read the PDF file
                          // and store it in a variable (e.g. Uint8List fileData)
                          // String fileString = utf8.decode(fileData);

                          try {
                            final pb = PocketBase('http://10.0.2.2:8090');
                            final record = await pb.collection('notes').create(
                              body: {
                                'subject':
                                    'Hello world!', // some regular text field
                              },
                              files: [
                                http.MultipartFile.fromString(
                                  'document',
                                  utf8.decode(fileData),
                                  filename: 'Blog.pdf',
                                ),
                              ],
                            ).then((record) {
                              print(record.id);
                              print(record.getStringValue('subject'));
                            });
                          } catch (e) {
                            print(e.toString());
                          }
                          // Open file picker and select file
                        },
                      ),
```

**Comment by ganigeorgiev** [comment]
@aryabyte21 

For the Flutter code, if you already have a bytes list, then you can use `http.MultipartFile.fromBytes('document', fileData, filename: 'Blog.pdf')`. For more information please refer to the http dartdoc - https://pub.dev/documentation/http/latest/http/MultipartFile-class.html (_it is also kindof strange to mix and chain both `await` and `then` and I'd advice to use only one or the other for a single `Future`_).

For the React Native code, I don't understand what the snippet is doing. `fileUri` and `getFile` doesn't seem to do anything. Are you sure that `pdfFile1` is actually set and instance of `File`?

Are there any errors thrown in any of the above? You can also start PocketBase with `--debug` and inspect the log messages in the console when making upload requests and/or check the logs in the Admin UI > Logs.

**Reply by aryabyte21** [reply]
@ganigeorgiev  I'm getting this error in flutter I/flutter (13957): FormatException: Missing extension byte (at offset 11)

**Reply by ganigeorgiev** [reply]
@aryabyte21 This looks similar to https://github.com/flutter/flutter/issues/81666. You can try applying the suggested workaround/fixes in the linked issue.

**Reply by ganigeorgiev** [reply]
But once again, you don't need to call `utf8.decode` (if that's when you are getting the above error) and you should be able to use directly `http.MultipartFile.fromBytes` just fine.

**Reply by aryabyte21** [reply]
It worked! Tqsm @ganigeorgiev I love pocketbase thanks for helping more power to you.
